ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreHardDelete.java

```diff
@@ -82,17 +82,20 @@ public MessageInfo getMessageInfo(Read read, long offset, StoreKeyFactory storeK
             return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
                 Utils.addSecondsToEpochTime(properties.getCreationTimeInMs(), properties.getTimeToLiveInSeconds()));
           } else {
-            boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(stream);
-            return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
-                deleteFlag);
+            DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(stream);
+            if (deleteRecord.getVersion() == MessageFormatRecord.Delete_Version_V1) {
+              return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(), true);
```

Comment:
why doesn't `stream.size()` work here? If it doesn't I would prefer exposing a function in delete record maybe to expose the size instead of hard coding.